src/exportJarSteps/ExportJarTaskProvider.ts

```diff
@@ -56,18 +126,168 @@ class ExportJarTaskTerminal implements Pseudoterminal {
         this.stepMetadata.elements = exportJarTaskDefinition.elements;
     }
 
-    public async open(initialDimensions: TerminalDimensions | undefined): Promise<void> {
-        await createJarFile(this.stepMetadata);
-        this.closeEmitter.fire();
+    public async open(_initialDimensions: TerminalDimensions | undefined): Promise<void> {
+        try {
+            if (this.stepMetadata.outputPath === undefined) {
+                // For the tasks whose targetPath is undefined, the user will select the output location manually.
+                this.stepMetadata.outputPath = "";
+            }
+            if (!_.isEmpty(this.stepMetadata.elements)) {
+                const runtimeOutputFolderMap: Map<string, string[]> = new Map<string, string[]>();
+                const runtimeArtifactMap: Map<string, string[]> = new Map<string, string[]>();
+                const testOutputFolderMap: Map<string, string[]> = new Map<string, string[]>();
+                const testArtifactMap: Map<string, string[]> = new Map<string, string[]>();
+                const projectList: INodeData[] = await Jdtls.getProjects(this.stepMetadata.workspaceFolder.uri.toString());
+                for (const project of projectList) {
+                    await this.setClasspathMap(project, "runtime", runtimeOutputFolderMap, runtimeArtifactMap);
+                    await this.setClasspathMap(project, "test", testOutputFolderMap, testArtifactMap);
+                }
+                this.stepMetadata.classpaths = await this.resolveClasspaths(runtimeOutputFolderMap,
+                    runtimeArtifactMap, testOutputFolderMap, testArtifactMap);
+            }
+            await createJarFile(this.stepMetadata);
+        } catch (err) {
+            if (err) {
+                failMessage(`${err}`);
+            }
+        } finally {
+            this.closeEmitter.fire();
+        }
     }
 
     public close(): void {
 
     }
-}
 
-interface IExportJarTaskDefinition extends TaskDefinition {
-    elements?: string[];
-    mainMethod?: string;
-    targetPath?: string;
+    private async setClasspathMap(project: INodeData, classpathScope: string,
+                                  outputFolderMap: Map<string, string[]>, artifactMap: Map<string, string[]>): Promise<void> {
+        const extensionApi: any = await getExtensionApi();
+        const classpathResult: IClasspathResult = await extensionApi.getClasspaths(project.uri, { scope: classpathScope });
+        const outputFolders: string[] = [];
+        const artifacts: string[] = [];
+        for (const classpath of classpathResult.classpaths) {
```

Comment:
for (const classpath of [...classpathResult.classpaths, ...classpathResult.modulepaths])